command/command.c

```diff
@@ -313,6 +349,30 @@ command_execute (CommandApplet *command_applet)
     return FALSE;
 }
 
+static gboolean
+command_launcher_execute (GtkWidget *widget, GdkEventButton *event, CommandApplet *command_applet)
+{
+    GError *error = NULL;
+    gchar *output = NULL;
+    gint ret = 0;
+    gchar *command_launcher;
+
+    /* Only react on left mouse click as we need right one for Preferences */
+    if (event->type == GDK_BUTTON_RELEASE && event->button == 1) {
+        command_launcher = g_settings_get_string (command_applet->settings, COMMAND_LAUNCHER_KEY);
```

Comment:
Why are you reading from GSettings again? Is not this already available as `command_applet->command_launcher`?